src/api/controllers/WalletController.js

```diff
@@ -52,6 +61,35 @@ class WalletController extends PointSDKController {
 
         return this._response({hash});
     }
+
+    async encryptData() {
+        const {publicKey, data} = this.payload;
+        const {host} = this.req.headers;
+        const encryptedData = await encryptData(host, data, publicKey);
+        return this._response(encryptedData);
+    }
+    
+    async decryptData() {
+        const {
+            encryptedData, 
+            encryptedSymmetricObj: unparsedEncryptedSymmetricObjJSON
+        } = this.payload;
+        const {host} = this.req.headers;
+        const privateKey = this.defaultWallet.privateKey;
+
+        const encryptedSymmetricObjJS = JSON.parse(unparsedEncryptedSymmetricObjJSON);
+        const encryptedSymmetricObj = {};
+        for (const k in encryptedSymmetricObjJS) {
+            encryptedSymmetricObj[k] = Buffer.from(encryptedSymmetricObjJS[k], 'hex');
+        }
```

Comment:
@nicobevilacqua is this logic useful elsewhere? Would it be better in the `encryption` module perhaps?

Reply:
ok